cyten/models/couplings.py

```diff
@@ -0,0 +1,757 @@
+        raise ValueError('Must have at least one non-zero prefactor.')
+    h = np.transpose(h, [0, 2, 3, 1])
+    return Coupling.from_dense_block(h, sites, name=name)
+
+
+def spin_field_coupling(sites: list[SpinDOF], hx: float = None, hy: float = None,
+                        hz: float = None, name: str = 'spin-field') -> Coupling:
+    """Coupling between a (single) spin and a (magnetic) field.
+
+    Parameters
+    ----------
+    hx, hy, hz : float, optional
+        If given, adds a corresponding term ``hx * S_i^x``, ``hy * S_i^y``, ``hz * S_i^z`` with
+        the value as prefactor.
+    """
+    # TODO test that this builds what we expect
+    assert len(sites) == 1
+    s = sites[0].spin_vector
+    h = 0
+    if hx is not None:
+        h += hx * s[:, :, 0]
+    if hy is not None:
+        h += hy * s[:, :, 1]
+    if hz is not None:
+        h += hz * s[:, :, 2]
+    if np.ndim(h) == 0:
+        raise ValueError('Must have at least one non-zero prefactor.')
+    return Coupling.from_dense_block(h, sites, name=name)
+
+
+def aklt_coupling(sites: list[SpinDOF], name: str = 'AKLT') -> Coupling:
+    r"""AKLT coupling between two spin-1 sites.
+    
+    Construct the AKLT coupling between S=1 spins as originally defined by
+    Affleck, Kennedy, Lieb, Tasaki in :cite:`affleck1987`, but drop the
+    constant part of 1/3 per bond and rescale with a factor of 2.
+
+    .. math ::
+        2 * P^{S=2}_{i,i+1} + const
+        = \vec{S}_i \cdot \vec{S}_{i+1}
+          + \frac{1}{3} (\vec{S}_i \cdot \vec{S}_{i+1})^2
+    """
+    # TODO test that this builds what we expect
+    assert len(sites) == 2
+    assert sites[0].double_total_spin == 2 == sites[1].double_total_spin
+    s1 = sites[0].spin_vector
+    s2 = sites[1].spin_vector
+    S_dot_S = np.tensordot(s1, s2, axes=[2, 2])
+    S_dot_S = np.transpose(S_dot_S, [0, 2, 3, 1])
+    S_dot_S_square = np.tensordot(S_dot_S, S_dot_S, axes=[[3, 2], [0, 1]])
+    h = S_dot_S + S_dot_S_square / 3.
+    return Coupling.from_dense_block(h, sites, name=name)
+
+
+def heisenberg_coupling(sites: list[SpinDOF], name: str = 'S.S') -> Coupling:
+    # TODO test that this builds what we expect
+    return spin_spin_coupling(sites=sites, xx=1, yy=1, zz=1, name=name)
+
+
+def chiral_3spin_coupling(sites: list[SpinDOF], name: str = 'S.SxS') -> Coupling:
+    # TODO test that this builds what we expect
+    assert len(sites) == 3
+    SxS = np.cross(sites[1].spin_vector[:, None, None, :, :],
+                   sites[2].spin_vector[None, :, :, None, :],
+                   axis=4)  # [p1, p2, p2*, p1*, i]
+    h = np.tensordot(sites[0].spin_vector, SxS, (-1, -1))  # [p0, p0*, p1, p2, p2*, p1*]
+    h = np.transpose(h, [0, 2, 3, 4, 5, 1])
+    return Coupling.from_dense_block(h, sites, name=name)
+
+
+# BOSON AND FERMION COUPLINGS
+
+def chemical_potential(sites: list[BosonicDOF | FermionicDOF], mu: float | list[float | None] = 1.,
+                       name: str = 'chem. pot.') -> Coupling:
```

Comment:
establish a convention for these names. either prefer pseudo-math or words. do we allow whitespace?